src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,60 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_STRING_PARAMETER_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNullParametersValue(user);
+        checkAge(user);
+        checkLogin(user);
+        checkPassword(user);
+        checkIfUserExists(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkNullParametersValue(User user) {
+        if (user == null) {
+            throw new InvalidDataException("User cant be null");
+        }
+        if (user.getAge() == null) {
+            throw new InvalidDataException("Age cant be null");
+        }
+        if (user.getPassword() == null) {
+            throw new InvalidDataException("Password cant be null");
+        }
+        if (user.getLogin() == null) {
+            throw new InvalidDataException("Login cant be null");
+        }
+    }
+
+    private void checkAge(User user) {
```

Comment:
```suggestion
    private void validateAge(User user) {
```
the same in all remaining methods